node/src/types/deploy.rs

```diff
@@ -46,6 +46,8 @@ use crate::{
     utils::DisplayIter,
 };
 
+const MAX_SERIALIZED_SIZE: u64 = 1_024 * 1_024;
```

Comment:
This should be part of the chainspec, rather than hardcoded. With this setup it's possible to run from the same chainspec with different binary that has this limit set differently. IMO this type of network-wide configuration should be introduced via chainspec, rather than software version.

Reply:
agreed